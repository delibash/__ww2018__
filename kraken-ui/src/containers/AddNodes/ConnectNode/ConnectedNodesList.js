import React from 'react';
import PropTypes from 'prop-types';
import { Label } from './../../../components/';
import { colors, SVGIcons } from './../../../global/constants';
import InlineSVG from 'svg-inline-react';
import styled from 'styled-components';

const StyledList = styled.ul`
  margin: 1.8rem 0;
`;

const StyledListItem = styled.li`
  margin-bottom: .8rem;
  display: flex;
`;

const StyledClose = styled.span`
  position: absolute;
  height: 4.2rem;
  width: 3rem;
  top: 0;
  right: 0;
  background-color: ${colors.LIGHTER_GRAY};
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  i {
    height: 1.5rem;
  }
`;

const ConnectedNodesList = props => {
  const {
    connectedNodes
  } = props;

  return (
    <StyledList>
      {connectedNodes && connectedNodes.map((val, i) => {
        return (
          <StyledListItem
            key={i}
          >
            <Label
              nodeLabel
            >
              <span>{val.value}</span>
              <StyledClose
                onClick={() => props.removeConnectedNode(i)}
              >
                <InlineSVG src={SVGIcons.close} />
              </StyledClose>
            </Label>
          </StyledListItem>
        );
      })}
    </StyledList>
  );
};

ConnectedNodesList.propTypes = {
  connectedNodes: PropTypes.arrayOf(PropTypes.object),
  nodeName: PropTypes.string,
  removeConnectedNode: PropTypes.func
};

export default ConnectedNodesList;