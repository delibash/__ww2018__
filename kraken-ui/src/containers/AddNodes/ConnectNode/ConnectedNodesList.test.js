import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import ConnectedNodesList from './ConnectedNodesList';

test('Renders ConnectedNodesList', () => {
  const tree = renderer.create(
    <ConnectedNodesList
      connectedNodes={[{value: '', label: ''}]}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});