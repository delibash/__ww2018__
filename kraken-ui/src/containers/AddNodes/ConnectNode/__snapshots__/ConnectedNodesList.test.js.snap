// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ConnectedNodesList 1`] = `
.c2 {
  padding: .6rem 1.2rem;
  font-size: 1.3rem;
  color: rgba(0,0,0,1);
  font-weight: 300;
  border: 0 none;
  border-radius: .2rem;
  background-color: rgba(255,255,255,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 4.2rem;
  width: 100%;
  box-shadow: 0 0 .5rem 0 rgba(0,0,0,0.5);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c0 {
  margin: 1.8rem 0;
}

.c1 {
  margin-bottom: .8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  position: absolute;
  height: 4.2rem;
  width: 3rem;
  top: 0;
  right: 0;
  background-color: rgba(234,236,242,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c3 i {
  height: 1.5rem;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <span
      className="c2"
      color="rgba(0, 0, 0, 1)"
      display="flex"
      fontWeight={300}
      height="4.2rem"
      width="100%"
    >
      <span>
        
      </span>
      <span
        className="c3"
        onClick={[Function]}
      >
        <i
          dangerouslySetInnerHTML={
            Object {
              "__html": "
  <svg viewBox=\\"0 0 20 20\\" height=\\"15px\\">
    <path
      fill=\\"black\\"
      d=\\"M15.898,4.045c-0.271-0.272-0.713-0.272-0.986,0l-4.71,4.711L5.493,4.045c-0.272-0.272-0.714-0.272-0.986,0s-0.272,0.714,0,0.986l4.709,4.711l-4.71,4.711c-0.272,0.271-0.272,0.713,0,0.986c0.136,0.136,0.314,0.203,0.492,0.203c0.179,0,0.357-0.067,0.493-0.203l4.711-4.711l4.71,4.711c0.137,0.136,0.314,0.203,0.494,0.203c0.178,0,0.355-0.067,0.492-0.203c0.273-0.273,0.273-0.715,0-0.986l-4.711-4.711l4.711-4.711C16.172,4.759,16.172,4.317,15.898,4.045z\\">
    </path>
  </svg>
  ",
            }
          }
          src={null}
        />
      </span>
    </span>
  </li>
</ul>
`;
